### PR TITLE
[serenity] Fix project to reflect upstream changes

### PR DESCRIPTION
**Pull request by DavidKorczynski**
The work of Serenity is now usptreamed.

The goal of this PR is to simplify the OSS-Fuzz files so we can do work of the future in thee Serenity repository itself. 

Cross-references:
https://github.com/SerenityOS/serenity/issues/4154
https://github.com/SerenityOS/serenity/pull/4176

The work in this PR also supersedes the PR here https://github.com/google/oss-fuzz/pull/4737 as we would like to remove the cmakelists.txt file from the OSS-Fuzz repo.

### CODE REVIEW

**Comment by DavidKorczynski** [comment]
Could you perhaps assist here @linusg

The recent changes in Serenity seem to have affected the Clang build, in particular:

```
/src/serenity/Meta/Lagom/../../AK/Vector.h:401:28: error: no matching constructor for initialization of 'regex::CompareTypeAndValuePair'                                                                                             
        new (slot(m_size)) T(forward<Args>(args)...);                                                       
                           ^ ~~~~~~~~~~~~~~~~~~~                                                       
/src/serenity/Libraries/LibRegex/RegexParser.cpp:1083:14: note: in instantiation of function template specialization 'AK::Vector<regex::CompareTypeAndValuePair, 0>::empend<regex::CharacterCompareType, unsigned long>' requested here
    compares.empend(CharacterCompareType::CharClass, (ByteCodeValueType)ch.value());                        
             ^                                                                                       
/src/serenity/Libraries/LibRegex/RegexByteCode.h:143:8: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 2 were provided                                                         
struct CompareTypeAndValuePair {                                                                       
       ^                                                                                                 
/src/serenity/Libraries/LibRegex/RegexByteCode.h:143:8: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 2 were provided                                                           
/src/serenity/Libraries/LibRegex/RegexByteCode.h:143:8: note: candidate constructor (the implicit default constructor)
not viable: requires 0 arguments, but 2 were provided                                                     
In file included from /src/serenity/Libraries/LibRegex/RegexParser.cpp:27:                       
In file included from /src/serenity/Libraries/LibRegex/RegexParser.h:29:                                    
In file included from /src/serenity/Libraries/LibRegex/RegexByteCode.h:29:                                
In file included from /src/serenity/Libraries/LibRegex/RegexMatch.h:32:                                   
In file included from /src/serenity/Meta/Lagom/../../AK/HashMap.h:32:                                        
/src/serenity/Meta/Lagom/../../AK/Vector.h:401:28: error: no matching constructor for initialization of 'regex::CompareTypeAndValuePair'                                                                                     
        new (slot(m_size)) T(forward<Args>(args)...);                                                    
                           ^ ~~~~~~~~~~~~~~~~~~~                                                    
/src/serenity/Libraries/LibRegex/RegexParser.cpp:1262:20: note: in instantiation of function template specialization 'AK::Vector<regex::CompareTypeAndValuePair, 0>::empend<regex::CharacterCompareType, regex::CharRange>' requested here
            ranges.empend(CharacterCompareType::CharRange, CharRange { first_atom.value().code_point, second_atom.value().code_point });                                                                                         
                   ^                                                                                     
/src/serenity/Libraries/LibRegex/RegexByteCode.h:143:8: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 2 were provided                                                           
struct CompareTypeAndValuePair {                                                                         
       ^                                                                                                          
/src/serenity/Libraries/LibRegex/RegexByteCode.h:143:8: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 2 were provided                                                         
/src/serenity/Libraries/LibRegex/RegexByteCode.h:143:8: note: candidate constructor (the implicit default constructor)
not viable: requires 0 arguments, but 2 were provided                                                       
In file included from /src/serenity/Libraries/LibRegex/RegexParser.cpp:27:                             
In file included from /src/serenity/Libraries/LibRegex/RegexParser.h:29:                                  
In file included from /src/serenity/Libraries/LibRegex/RegexByteCode.h:29:                                  
In file included from /src/serenity/Libraries/LibRegex/RegexMatch.h:32:                      
In file included from /src/serenity/Meta/Lagom/../../AK/HashMap.h:32:                              
/src/serenity/Meta/Lagom/../../AK/Vector.h:401:28: error: no matching constructor for initialization of 'regex::CompareTypeAndValuePair'                                                                                 
        new (slot(m_size)) T(forward<Args>(args)...);                                            
                           ^ ~~~~~~~~~~~~~~~~~~~                                             
/src/serenity/Libraries/LibRegex/RegexParser.cpp:1274:20: note: in instantiation of function template specialization 'AK::Vector<regex::CompareTypeAndValuePair, 0>::empend<regex::CharacterCompareType, unsigned int &>' requested here
            ranges.empend(CharacterCompareType::Char, first_atom.value().code_point);                           
                   ^                                                                                   
/src/serenity/Libraries/LibRegex/RegexByteCode.h:143:8: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 2 were provided                                                
struct CompareTypeAndValuePair {                                                               
       ^                                             
```

**Comment by linusg** [comment]
I believed this is already being looked into, here: https://github.com/SerenityOS/serenity/pull/4181

**Comment by DavidKorczynski** [comment]
That's awesome - am so impressed with the efficiency of all you Serenity maintainers. 

Okay, I will keep looking out for that one. When that one is merged this one should be good to go - the set up in this PR worked prior to the Regex additions.

**Review by inferno-chromium — APPROVED** [review]
Thank you so much!

File: projects/serenity/project.yaml [hunk 1]
```diff
@@ -3,6 +3,9 @@ language: c++
 primary_contact: "kling@serenityos.org"
 auto_ccs:
   - "david@adalogics.com"
+  - "mail@linusgroh.de"
+  - "ali.mpfard@gmail.com"
+  - "luke.wilde@live.co.uk"
```
Comment (bugaevc) [inline projects/serenity/project.yaml:8]: Please add me too :slightly_smiling_face:
`"bugaevc@serenityos.org"`

**Pull request merged**